code/__DEFINES/is_helpers.dm

```diff
@@ -240,3 +240,5 @@ GLOBAL_LIST_INIT(glass_sheet_types, typecacheof(list(
 #define isshuttleturf(T) (length(T.baseturfs) && (/turf/baseturf_skipover/shuttle in T.baseturfs))
 
 #define isProbablyWallMounted(O) (O.pixel_x > 20 || O.pixel_x < -20 || O.pixel_y > 20 || O.pixel_y < -20)
+
+#define isfinite(n) (isnum(n) && n == n)
```

Comment:
This doesn't actually work.
```suggestion
#define isfinite(n) (x != 1.#IND && abs(x) < 1.#INF)
```
Handles NaN, positive infinity, and negative infinity.